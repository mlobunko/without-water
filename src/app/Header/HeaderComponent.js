import React from 'react';
import { LinkContainer } from 'react-router-bootstrap';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';

export const HeaderComponent = ({
  authorized,
  authorizedActionsLogOut,
  isOpen,
  toggle
}) => (
  <Navbar color="dark" dark expand="md">
    <LinkContainer to="/">
      <NavbarBrand className="mr-auto">
        <div className="nav-bar__logo" />
      </NavbarBrand>
    </LinkContainer>
    <NavbarToggler onClick={toggle} className="mr-2" />
    <Collapse isOpen={isOpen} navbar>
      <Nav className="ml-auto" navbar>
        <LinkContainer to="/news" activeClassName="active">
          <NavItem>
            <NavLink className="nav-bar__link">News</NavLink>
          </NavItem>
        </LinkContainer>
        <LinkContainer to="/profile" activeClassName="active">
          <NavItem>
            <NavLink className="nav-bar__link">Profile</NavLink>
          </NavItem>
        </LinkContainer>
        {authorized ? (
          <LinkContainer exact to="/">
            <NavItem onClick={authorizedActionsLogOut}>
              <NavLink className="nav-bar__link">Log out</NavLink>
            </NavItem>
          </LinkContainer>
        ) : (
          <LinkContainer to="/login" activeClassName="active">
            <NavItem>
              <NavLink className="nav-bar__link">Log in</NavLink>
            </NavItem>
          </LinkContainer>
        )}
      </Nav>
    </Collapse>
  </Navbar>
);
