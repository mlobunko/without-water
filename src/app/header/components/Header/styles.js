import styled from 'styled-components';
import { NavbarBrand, NavLink } from 'reactstrap';
import imageSieve from '../../images/sieve.svg';

export const StyledNavbarBrand = styled(NavbarBrand)`
  background: url(${imageSieve}) no-repeat center center;
  background-size: contain;
  height: 2.5rem;
  width: 2.5rem;
`;

export const StyledNavLink = styled(NavLink)`
  &:hover {
    cursor: pointer;
  }
`;
