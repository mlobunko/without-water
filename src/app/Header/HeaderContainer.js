import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import authorized from 'shared/authorized';
import { HeaderComponent } from './HeaderComponent';

export class HeaderContainer extends React.Component {
  state = {
    isOpen: false
  };

  handleAddRemoveEventListener = () => {
    if (this.state.isOpen) {
      document.addEventListener('click', this.handleDocumentClick);
    } else {
      document.removeEventListener('click', this.handleDocumentClick);
    }
  };

  toggle = () => {
    this.setState(
      {
        isOpen: !this.state.isOpen
      },
      this.handleAddRemoveEventListener
    );
  };

  handleDocumentClick = e => {
    const container = this.element;
    if (e.target !== container && !container.contains(e.target)) {
      this.toggle();
    }
  };

  componentWillUnmount = () => {
    document.removeEventListener('click', this.handleDocumentClick);
  };

  render() {
    return (
      <div ref={element => (this.element = element)}>
        <HeaderComponent
          isOpen={this.state.isOpen}
          logOut={this.props.logOut}
          toggle={this.toggle}
          authorized={this.props.authorized}
          authorizedActionsLogOut={this.props.authorizedActionsLogOut}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  authorized: state.authorized
});

const mapDispatchToProps = {
  authorizedActionsLogOut: authorized.actions.logOut
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(HeaderContainer)
);
