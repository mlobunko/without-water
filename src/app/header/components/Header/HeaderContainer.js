import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import HeaderComponent from './HeaderComponent';

import authorized from 'shared/authorized';

export class HeaderContainer extends React.Component {
  state = {
    isOpen: false
  };

  handleAddRemoveEventListener = () => {
    if (this.state.isOpen) {
      document.addEventListener('click', this.handleDocumentClick);
    } else {
      document.removeEventListener('click', this.handleDocumentClick);
    }
  };

  toggle = () => {
    this.setState(
      {
        isOpen: !this.state.isOpen
      },
      this.handleAddRemoveEventListener
    );
  };

  handleDocumentClick = e => {
    const container = this.element;
    if (e.target !== container && !container.contains(e.target)) {
      this.toggle();
    }
  };

  componentWillUnmount = () => {
    document.removeEventListener('click', this.handleDocumentClick);
  };

  render() {
    return (
      <div ref={element => (this.element = element)}>
        <HeaderComponent
          isOpen={this.state.isOpen}
          logOut={this.props.logOut}
          toggle={this.toggle}
          authorized={this.props.authorized}
          authorizedActionsLogOut={this.props.authorizedActionsLogOut}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  authorized: authorized.selectors.getAll(state)
});

const mapDispatchToProps = {
  authorizedActionsLogOut: authorized.actions.logOut
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(HeaderContainer)
);

HeaderContainer.propTypes = {
  authorized: PropTypes.bool
};
