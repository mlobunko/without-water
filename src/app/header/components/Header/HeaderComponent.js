import React from 'react';
import { LinkContainer } from 'react-router-bootstrap';
import { Collapse, Navbar, NavbarToggler, Nav, NavItem } from 'reactstrap';
import PropTypes from 'prop-types';

import { StyledNavbarBrand, StyledNavLink } from './styles';

export const HeaderComponent = ({
  authorized,
  authorizedActionsLogOut,
  isOpen,
  toggle
}) => (
  <Navbar color="dark" dark expand="md">
    <LinkContainer to="/">
      <StyledNavbarBrand />
    </LinkContainer>
    <NavbarToggler onClick={toggle} className="mr-2" />
    <Collapse isOpen={isOpen} navbar>
      <Nav className="ml-auto" navbar>
        <LinkContainer to="/news" activeClassName="active">
          <NavItem>
            <StyledNavLink>News</StyledNavLink>
          </NavItem>
        </LinkContainer>
        <LinkContainer to="/profile" activeClassName="active">
          <NavItem>
            <StyledNavLink>Profile</StyledNavLink>
          </NavItem>
        </LinkContainer>
        {authorized ? (
          <LinkContainer exact to="/">
            <NavItem onClick={authorizedActionsLogOut}>
              <StyledNavLink>Log out</StyledNavLink>
            </NavItem>
          </LinkContainer>
        ) : (
          <LinkContainer to="/login" activeClassName="active">
            <NavItem>
              <StyledNavLink>Log in</StyledNavLink>
            </NavItem>
          </LinkContainer>
        )}
      </Nav>
    </Collapse>
  </Navbar>
);

export default HeaderComponent;

HeaderComponent.propTypes = {
  authorized: PropTypes.bool.isRequired,
  authorizedActionsLogOut: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired
};
