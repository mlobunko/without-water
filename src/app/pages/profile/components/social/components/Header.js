import React from 'react';
import PropTypes from 'prop-types';

import { StyledHeader } from './styles';

export const Header = ({ text }) => <StyledHeader>{text}</StyledHeader>;

Header.displayName = 'Header';

export default Header;

Header.propTypes = {
  text: PropTypes.string.isRequired
};
