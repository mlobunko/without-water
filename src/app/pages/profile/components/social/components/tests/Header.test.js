import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../Header';

describe('render', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header text="London" />);
    expect(wrapper).toMatchSnapshot();
  });
});
