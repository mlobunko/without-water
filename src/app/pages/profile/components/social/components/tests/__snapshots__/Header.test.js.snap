// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<styled.h5>
  London
</styled.h5>
`;
